### PR TITLE
Pumpkinpeople - A Halloween special

### PR DESCRIPTION
**Pull request by DrDuckedGoose**
## About The Pull Request

Another feature PR, I'm sorry. It's just some hacktober fun, this should be the 4th & final PR for that. 

Adds a new Halloween-only race, pumpkin people. This race is essentially a re-flavored podpeople for the spooky,  and silly, holiday. They have the additional ability to pull candy from their head. Pulling too much candy out at once will cause the player to pull their brain out.

## Why It's Good For The Game

Podpeople are a fun option for botanists, among other jobs. Lets people experiment with a new species RP over Halloween.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

image
image
image
image
image

image
image
image

</details>

## Changelog
:cl:
add: Adds ``pumpkin_man`` species for Halloween
imageadd: Adds ``pumkin_man`` limbs & respective sprites
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
They should be magic mirror restricted race if I remember our standards correctly.

**Comment by DrDuckedGoose** [comment]
dullahan flags
``changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_PRIDE | MIRROR_MAGIC | ERT_SPAWN``
vs podpeople flags
``changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_MAGIC | MIRROR_PRIDE | RACE_SWAP | ERT_SPAWN | SLIME_EXTRACT``

**Comment by PowerfulBacon** [comment]
We should just reskin podpeople on halloween to this, rather than have a new species

**Comment by DrDuckedGoose** [comment]
This means we can have it every halloween though? 
Maybe it's bloat but, surely not enough to hurt?

**Comment by PowerfulBacon** [comment]
I mean, I don't see a reason not to, but I also don't see a reason to

**Comment by dwasint** [comment]
doesn't give them pumpkin juice blood 0/10

**Comment by PowerfulBacon** [comment]
These can be added during Halloween but won't be a permanent merge

**Comment by DrDuckedGoose** [comment]
No worries, I just need the approval check

**Comment by Crossedfall** [comment]
It's time

**Comment by WrillWasTaken** [comment]
Shouldn't their head-candy regenerate over time? Once you use it all up the pull candy button becomes an instant-death button for the rest of the round.

**Comment by DrDuckedGoose** [comment]
> Shouldn't their head-candy regenerate over time? Once you use it all up the pull candy button becomes an instant-death button for the rest of the round.

It already should, in the code. However, I made a mistake and left ``. = ..()`` in the ``process`` which breaks it.
This has been fixed, thank you for notifying me.

File: code/modules/mob/living/carbon/human/species_types/pumpkin_man.dm [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	species_l_arm = /obj/item/bodypart/l_arm/pumpkin_man
+	species_r_arm = /obj/item/bodypart/r_arm/pumpkin_man
+	species_l_leg = /obj/item/bodypart/l_leg/pumpkin_man
+	species_r_leg = /obj/item/bodypart/r_leg/pumpkin_man
+
+//Only allow race roundstart on Halloween
+/datum/species/pod/pumpkin_man/check_roundstart_eligible()
+	if(SSevents.holidays && SSevents.holidays[HALLOWEEN])
+		return TRUE
+	return ..()
```
Comment (Crossedfall) [inline code/modules/mob/living/carbon/human/species_types/pumpkin_man.dm]:
```suggestion
	return FALSE
```

**Comment by Crossedfall** [comment]
These went over very well. I think it would actually be fine to make them a proper addition that only shows up during Halloween.

File: code/modules/mob/living/carbon/human/species_types/pumpkin_man.dm (hunk 1 above)
Comment (Crossedfall) [reply code/modules/mob/living/carbon/human/species_types/pumpkin_man.dm]: Should default to `FALSE` if the intent is to actually make it Halloween only.

**Review by Crossedfall — APPROVED** [review]

**Comment by Crossedfall** [comment]
Stuck checks moment.

**Comment by Crossedfall** [comment]
@PowerfulBacon Did you decide on if we wanted to merge this in or not? It's set to only enable during the Halloween event, which seems fine to me given how positively they were received.

**Comment by PowerfulBacon** [comment]
I'm fine with merge if they are only turned on in Halloween. It's self contained so not causing disastrous bloat

**Pull request merged**